src/snowflake/cli/_app/telemetry.py

```diff
@@ -83,6 +88,43 @@ class TelemetryEvent(Enum):
 TelemetryDict = Dict[Union[CLITelemetryField, TelemetryField], Any]
 
 
+def _is_cli_exception(exception: Exception) -> bool:
+    return isinstance(
+        exception,
+        (
+            click.ClickException,
+            typer.Exit,
+            typer.Abort,
+            BrokenPipeError,
+            KeyboardInterrupt,
+        ),
+    )
+
+
+def _get_additional_exception_information(exception: Exception) -> TelemetryDict:
+    """
+    Attach the errno and sqlstate if the exception or the
+    cause of the exception is a ProgrammingError
+    """
+    additional_info = {}
+
+    if isinstance(exception, ProgrammingError):
+        additional_info[CLITelemetryField.ERROR_CODE] = exception.errno
+        additional_info[CLITelemetryField.SQL_STATE] = exception.sqlstate
+
+    if exception.__cause__:
+        cause = exception.__cause__
+        additional_info[CLITelemetryField.ERROR_CAUSE] = type(cause).__name__
```

Comment:
what if we don't have a cause, don't we want to fallback to using the exception type itself?

Reply:
Is already being tracked, see https://github.com/snowflakedb/snowflake-cli/pull/1651#discussion_r1805238140